src/gui/qgsmaptoolzoom.cpp

```diff
@@ -85,6 +85,7 @@ void QgsMapToolZoom::canvasPressEvent( QgsMapMouseEvent *e )
     return;
 
   mZoomRect.setRect( 0, 0, 0, 0 );
+  mZoomRect.setTopLeft( e->pos() );
```

Comment:
Why is this needed?
This will set a rect from 0,0 to the mouse pos.
If anything, we could also set bottom right to mouse pos so that we're starting with an empty rect at mouse click pos.

Reply:
Ah, that was my intention but missed changing the line - thanks for catching it! We now start with an empty rect as mouse click pos.